servicetalk-test-resources/src/main/resources/junit-platform.properties

```diff
@@ -1,5 +1,8 @@
+#
+# See https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-parallel-execution
+#
 junit.jupiter.execution.parallel.enabled = true
 junit.jupiter.execution.parallel.mode.default = same_thread
-junit.jupiter.execution.parallel.mode.classes.default = concurrent
+junit.jupiter.execution.parallel.mode.classes.default = same_thread
```

Comment:
having `same_thread` on both won't do anything concurrently, see the picture here: https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-parallel-execution
What you had previously was the right choice. It executes all tests withing a single class sequentially, but runs independent test classes concurrently:
```
junit.jupiter.execution.parallel.enabled = true
junit.jupiter.execution.parallel.mode.default = same_thread
junit.jupiter.execution.parallel.mode.classes.default = concurrent
```

Reply:
As proposed, the options are intended to run tests in the same manner as our current non-parallel execution. Only the first line, `junit.jupiter.execution.parallel.enabled = true`, is intended to have any effect. The rest of the lines just make the defaults explicit (I hope, though it is certainly possible I screwed it up). So the effect of the first line is minor in that it merely allows parallel execution, it doesn't cause parallel execution. This configuration works with our current tests, some of which fail when executed in parallel due to interdependencies between cases, but does allow us to start adding `@Execution` annotations to individual suites/tests that can be parallelized. There are some rather large suites parameterized test that can hopefully be accelerated.

Reply:
Hi all - I found my way here from a [thread](https://gitter.im/junit-team/junit5?at=60f721a2926ce249e576a3b1) started on the JUnit 5 Gitter.

I'm rather curious, have you already tried running the tests with all the `parallel.mode` sub-options set to `concurrent`?

Either way, I completely understand why you're taking the cautious approach. :)

Reply:
@jbduncan wrote:
> I'm rather curious, have you already tried running the tests with all the `parallel.mode` sub-options set to `concurrent`?

Yes, we have tried `concurrent` as the default and some of our tests do fail. We will have to incrementally mark tests using `@Execution(CONCURRENT)` and work on fixing individual tests. There are a few "tentpole" tests that consume a lot of time due to sequential execution (and hold up subsequent parallel portions of the build) so we will probably focus on those first.